sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -502,33 +526,59 @@ def __init__(self, max_size_in_bytes=None, partition_id=None, partition_key=None
                 "partition_key to only be string type, they might fail to parse the non-string value."
             )
 
-        self.max_size_in_bytes = max_size_in_bytes or constants.MAX_FRAME_SIZE_BYTES
-        self.message = BatchMessage(data=[])
         self._partition_id = partition_id
         self._partition_key = partition_key
-        self.message = set_message_partition_key(self.message, self._partition_key)
-        self._size = pyutils.get_message_encoded_size(self.message)
-        self._count = 0
 
-    def __repr__(self):
-        # type: () -> str
-        batch_repr = "max_size_in_bytes={}, partition_id={}, partition_key={!r}, event_count={}".format(
-            self.max_size_in_bytes, self._partition_id, self._partition_key, self._count
+        self._message = self._amqp_transport.build_batch_message(data=[])
+        self._message = self._amqp_transport.set_message_partition_key(
+            self._message, self._partition_key
+        )
+        self._size = self._amqp_transport.get_batch_message_encoded_size(self._message)
+        self.max_size_in_bytes = (
+            max_size_in_bytes or self._amqp_transport.MAX_MESSAGE_LENGTH_BYTES
         )
-        return "EventDataBatch({})".format(batch_repr)
 
-    def __len__(self):
+        self._count = 0
+        self._internal_events: List[
+            Union[EventData, AmqpAnnotatedMessage]
+        ] = []
+        self._uamqp_message = None if PyamqpTransport.TIMEOUT_FACTOR == self._amqp_transport.TIMEOUT_FACTOR else self._message
```

Comment:
If PyamqpTransport, then we want to construct a LegacyBatchMessage. Otherwise, I set this to the uamqp BatchMessage.